dynamic/provider_test.go

```diff
@@ -546,3 +553,81 @@ func must[T any](v T, err error) T {
 func marshal(m resource.PropertyMap) *structpb.Struct {
 	return must(plugin.MarshalProperties(m, plugin.MarshalOptions{}))
 }
+
+func TestLogReplayProvider(t *testing.T) {
+	grpcLogs, err := os.ReadFile("./testdata/TestLogReplayProvider/grpc_log_random.json")
+	if err != nil {
+		t.Fatalf("failed to read grpc log: %v", err)
+	}
+
+	provPlugin := grpcutil.NewLogReplayProvider("random", "0.0.1", "url", string(grpcLogs))
+	prov := v6shim.New(provPlugin)
+	require.NoError(t, err)
+
+	resp, err := prov.GetProviderSchema(context.Background(), &tfprotov6.GetProviderSchemaRequest{})
+	require.NoError(t, err)
+	require.Contains(t, resp.ResourceSchemas, "random_bytes")
+	require.Contains(t, resp.ResourceSchemas, "random_pet")
+	require.Contains(t, resp.ResourceSchemas, "random_string")
+
+	configVal, err := tfprotov6.NewDynamicValue(
+		tftypes.Object{},
+		tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{}),
+	)
+	require.NoError(t, err)
+	configResp, err := prov.ValidateProviderConfig(context.Background(), &tfprotov6.ValidateProviderConfigRequest{
+		Config: &configVal,
+	})
+
+	require.NoError(t, err)
+	require.Equal(t, "\x80", string(configResp.PreparedConfig.MsgPack))
+}
+
+type provider struct {
+	tfprotov6.ProviderServer
+	io.Closer
+
+	name    string
+	version string
+	url     string
+}
+
+func (p provider) Name() string    { return p.name }
+func (p provider) Version() string { return p.version }
+func (p provider) URL() string     { return p.url }
+
+func TestLogReplayProviderWithProgram(t *testing.T) {
```

Comment:
What does it mean to use the LogReplayProvider with a program? What is the expected behavior?